src/test/java/bowling/FrameTest.java

```diff
@@ -0,0 +1,225 @@
+package bowling;
+
+import bowling.exception.BowlingException;
+import bowling.exception.FrameOverException;
+import bowling.exception.InvalidPinQuantityException;
+import org.junit.Test;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class FrameTest {
+    @Test
+    public void normalFrameHasTwoRolls() throws BowlingException {
```

Comment:
Usually when writing tests I do the following:
* Name the method as: public void methodYouAreTesting_When<PreCondition>_Then<Assertion>
* public void roll_WhenFrameHasTwoRolls_ThenGameIsOver

Reply:
Yes that's a good approach and I have seen it used successfully in many places.

Personally I try to keep things short. You are testing the class and it's behaviour/properties. So I tend to just describe the property
`gamesCanBeStarted`
`framesCanBeCompleted`
`invalidOperationOnStrikeFrameThrowsException`

But of course, if these properties are exhibited by many methods then that can cause confusion. In which case the method should be questioned - why is it public and how is it different?